### PR TITLE
unit testcase for checkpoint validation during create data stream

### PR DESCRIPTION
**Pull request by sureshdash2022-yb**
please review those testcases
@suranjan @vaibhav-yb @isha-yb

### CODE REVIEW

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc [hunk 1]
```diff
@@ -463,6 +462,67 @@ TEST_F(CDCSDKStreamTest, YB_DISABLE_TEST_IN_TSAN(CDCWithXclusterEnabled)) {
...
+  //get the list of dbstream
+  google::protobuf::RepeatedPtrField<yb::master::CDCStreamInfoPB> list_streams =
+      ASSERT_RESULT(ListDBStreams(kNamespaceName));
+  const uint32_t num_streams = list_streams.size();
+
+  for (uint32_t i = 0; i < num_streams; ++i) {
+    //validate the streamid
+    ASSERT_EQ(db_stream_id, list_streams.Get(i).stream_id());
+
+    uint32_t options_sz = list_streams.Get(i).options_size();
```
Comment (vaibhav-yb) [inline ent/src/yb/integration-tests/cdcsdk_stream-test.cc:484]: `const uint32_t options_sz`

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc [hunk 2]
```diff
@@ -463,6 +462,67 @@ TEST_F(CDCSDKStreamTest, YB_DISABLE_TEST_IN_TSAN(CDCWithXclusterEnabled)) {
...
+    for (uint32_t j = 0; j < options_sz; j++) {
+      //validate the checkpoint type IMPLICIT
+      string curKey = list_streams.Get(i).options(j).key();
```
Comment (vaibhav-yb) [inline ent/src/yb/integration-tests/cdcsdk_stream-test.cc:487]: In cpp, the we follow snake casing so the variable would be `cur_key` and `cur_value` in the next line.

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc [hunk 3]
```diff
@@ -463,6 +462,67 @@ TEST_F(CDCSDKStreamTest, YB_DISABLE_TEST_IN_TSAN(CDCWithXclusterEnabled)) {
...
+        //validate the checkpoint type IMPLICIT
+        string curKey = list_streams.Get(i).options(j).key();
+        string curValue = list_streams.Get(i).options(j).value();
+        if (curKey == string("checkpoint_type")) {
+          ASSERT_EQ(curValue, string("EXPLICIT"));
+        }
+      }
+    }
+}
+
```
Comment (vaibhav-yb) [inline ent/src/yb/integration-tests/cdcsdk_stream-test.cc:525]: Same changes are required here
* snake casing in variable names
* Writing comments correctly

**Review by vaibhav-yb — CHANGES_REQUESTED** [review]

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc (hunk 3 above)
Comment (sureshdash2022-yb) [reply ent/src/yb/integration-tests/cdcsdk_stream-test.cc:525]: Fixed

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc (hunk 2 above)
Comment (sureshdash2022-yb) [reply ent/src/yb/integration-tests/cdcsdk_stream-test.cc]: Fixed

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc (hunk 1 above)
Comment (sureshdash2022-yb) [reply ent/src/yb/integration-tests/cdcsdk_stream-test.cc]: Fixed

File: ent/src/yb/integration-tests/cdcsdk_stream-test.cc [hunk 4]
```diff
@@ -463,6 +462,67 @@ TEST_F(CDCSDKStreamTest, YB_DISABLE_TEST_IN_TSAN(CDCWithXclusterEnabled)) {
...
+TEST_F(CDCSDKStreamTest, YB_DISABLE_TEST_IN_TSAN(ImplicitCheckPointValidate)) {
+
+  // Create a cluster.
+  ASSERT_OK(SetUpWithParams(3, 1, false));
+
+  // Create a DB Stream
+  std::string db_stream_id = ASSERT_RESULT(CreateDBStream(CDCCheckpointType::IMPLICIT));
+  ASSERT_NE(0, db_stream_id.length());
+
+  //get the list of dbstream
```
[the remark this answers is not in the document]
Comment (sureshdash2022-yb) [reply ent/src/yb/integration-tests/cdcsdk_stream-test.cc]: Fixed

**Comment by sureshdash2022-yb** [comment]
@vaibhav-yb @suranjan @isha-yb  fixed those review comments

**Review by suranjan — APPROVED** [review]

**Pull request merged**